app/utility/op_control.py

```diff
@@ -0,0 +1,51 @@
+from enum import Enum
+import asyncio
+
+class OpState(Enum):
+    RUN = "RUNNING"
+    PAUSE = "PAUSED"
+    CANCEL = "CANCELED"
+
+class OpControl:
```

Comment:
i like this class a lot. but i don't think i love it being a utility class, because it is reliant on the DAO (i'm aiming to keep the utility modules to be mostly static/1-off modules not reliant on the DB, whereas the services are the "connected layer". can we make this a base class of operation_svc instead?